aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -901,6 +901,95 @@ Tensor _cholesky_helper_cuda(const Tensor& self, bool upper) {
   }
 }
 
+template <typename scalar_t>
+static void apply_cholesky_mod(const Tensor& self, Tensor& out, Tensor& err, bool upper, std::vector<int64_t>& infos) {
+#ifndef USE_MAGMA
+AT_ERROR("cholesky: MAGMA library not found in "
+    "compilation. Please rebuild with MAGMA.");
+#else
+  magma_uplo_t uplo = upper ? MagmaUpper : MagmaLower;
+  Tensor outTag =  out;
+  Tensor selfTag = self;
+
+  auto self_matrix_stride = matrixStride(self);
+  auto batch_size = batchCount(self);
+  auto n = self.size(-2);
+  auto m = self.size(-1);
+
+  if (batch_size > 1) {
+    outTag =  out.flatten(0, -3);
+    selfTag = self.flatten(0, -3);
+  }
+  
+  scalar_t* self_data = selfTag.data_ptr<scalar_t>();
+  scalar_t* out_data = outTag.data_ptr<scalar_t>();
+  int info;
+  /* TODO: do this in parallel */
+  for (int64_t i = 0; i < batch_size; i++) {
```

Comment:
This is probably not ideal. What could be tried instead is: run `magmaCholeskyBatched` to get the list of infos pertaining to the batch, and then filter out the non-PD matrices, and add jitter to them alone. This method, according to me, is not leveraging MAGMA's parallelism capabilities.

Reply:
Same as above, if on CPU, makes no sense to parallelize because of the underlying parallelization in Lapack, and if on CUDA, this serial host loop will enqueue the CUDA stream queues.